CMakeLists.txt

```diff
@@ -51,12 +51,20 @@ FIND_PACKAGE(PkgConfig)
 FIND_PACKAGE(Threads REQUIRED)
 
 IF ((CMAKE_SYSTEM_NAME STREQUAL "Darwin") AND NOT (DEFINED OPENSSL_ROOT_DIR OR DEFINED ENV{OPENSSL_ROOT_DIR}))
-    MESSAGE(STATUS "*************************************************************************\n"
-                "   * Setting OPENSSL_ROOT_DIR to /usr/local/opt/openssl. On macOS, OpenSSL *\n"
-                "   * should be installed using homebrew or OPENSSL_ROOT_DIR must be set to *\n"
-                "   * the path that has OpenSSL installed.                                  *\n"
-                "   *************************************************************************")
-    SET(OPENSSL_ROOT_DIR "/usr/local/opt/openssl")
+    EXECUTE_PROCESS(
+        COMMAND brew --prefix openssl
+        OUTPUT_VARIABLE OPENSSL_ROOT_DIR
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_QUIET
+    )
```

Comment:
What happens if openssl is not installed?

Can the user understand that we are expecting openssl to be installed via brew but that it could not be found?

Reply:
This PR does not change what macOS users do. The message below describes what it is, even with this PR.

I have no idea whether `SET(OPENSSL_ROOT_DIR, "/usr/local/opt/openssl")` works or not, though.

Reply:
My question is if the warning that we print now would continue to be printed when openssl is *not* installed via brew. 

It has to be printed in such case, because it would become difficult for people to find out what the assumption is.

Reply:
OMG `brew --prefix FORMULA` retturns a correct path even if the formula is not installed. My method, `brew --prefix hogehoge`, was wrong 😭 .

Fixed bed796d12d829a30402cf8d243913d3b17cc5079